test/Microsoft.AspNet.Mvc.ModelBinding.Test/Metadata/DefaultModelMetadataProviderTest.cs

```diff
@@ -86,6 +86,7 @@ public void GetMetadataForProperties_Cached()
             // Assert
             for (var i = 0; i < metadata1.Length; i++)
             {
+                Assert.Same(metadata1[0], metadata2[1]);
```

Comment:
actually, why is this assertion valid?  the two properties have different names, attributes, and more